### PR TITLE
KevinDepuydt/missing-credentials-exceptions

### PR DESCRIPTION
**Pull request by KevinDepuydt**
- added `checkEnvironmentVariables` to utils/environment
- removed `Bot.checkEnvVars`
- displays clearer errors in train and run scripts
- tests `checkEnvironmentVariables`

### CODE REVIEW

**Comment by KevinDepuydt** [comment]
@windkomo the app token is also required for identifying bots sharing a database, maybe I could throw an exception only in these cases.
What do you think about it @yangeorget ?

File: tests/utils/environment.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+const buildConfig = configPart => Object.assign({}, defaultConfig, configPart);
+
+describe('Environment utils', () => {
+  describe('When BOTFUEL_APP_TOKEN is not defined', () => {
+    beforeAll(() => {
+      delete process.env.BOTFUEL_APP_TOKEN;
+    });
+
+    test('should throw an error when using botfuel adapter', async () => {
+      const config = buildConfig({ adapter: 'botfuel' });
```
Comment (yangeorget) [inline tests/utils/environment.test.js]: can't you just write const config = { ...defaultConfig, adapter: 'botfuel' };
 ?

File: tests/utils/environment.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      const f = () => checkCredentials(config);
+      expect(f).toThrowError(MissingCredentialsError);
+    });
+
+    test('should throw an error when using mongo brain', async () => {
+      const config = buildConfig({ brain: 'mongo' });
+      const f = () => checkCredentials(config);
```
Comment (yangeorget) [inline tests/utils/environment.test.js]: you could give a more explicit name such as failingFunction instead of f

**Comment by yangeorget** [comment]
LGTM ... added a couple of comments in the tests (feel free not to apply them if you don't agree)

File: tests/utils/environment.test.js (hunk 2 above)
Comment (KevinDepuydt) [reply tests/utils/environment.test.js]: I simplified this by using `expect(() => checkCredentials(config))` directly

File: tests/utils/environment.test.js (hunk 1 above)
Comment (KevinDepuydt) [reply tests/utils/environment.test.js]: Yes, I simplified tests

**Pull request merged**